modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/loader/MuleSdkPluginExtensionModelLoader.java

```diff
@@ -0,0 +1,111 @@
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.mule.internal.loader;
+
+import static org.mule.runtime.ast.api.ArtifactType.MULE_EXTENSION;
+import static org.mule.runtime.ast.api.util.MuleAstUtils.validatorBuilder;
+import static org.mule.runtime.ast.api.xml.AstXmlParser.builder;
+import static org.mule.runtime.extension.api.ExtensionConstants.MULE_SDK_ARTIFACT_AST_PROPERTY_NAME;
+import static org.mule.runtime.extension.api.ExtensionConstants.MULE_SDK_EXTENSION_LOADER_ID;
+import static org.mule.runtime.extension.api.ExtensionConstants.MULE_SDK_RESOURCE_PROPERTY_NAME;
+import static org.mule.runtime.module.artifact.activation.internal.ast.ArtifactAstUtils.parseArtifact;
+import static org.mule.runtime.module.artifact.activation.internal.ast.validation.AstValidationUtils.logWarningsAndThrowIfContainsErrors;
+import static org.mule.runtime.module.extension.mule.internal.loader.parser.utils.MuleSdkExtensionLoadingUtils.getRequiredLoadingParameter;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.ast.api.ArtifactAst;
+import org.mule.runtime.ast.api.xml.AstXmlParser;
+import org.mule.runtime.core.api.config.ConfigurationException;
+import org.mule.runtime.extension.api.loader.ExtensionLoadingContext;
+import org.mule.runtime.extension.api.loader.ExtensionModelLoader;
+import org.mule.runtime.extension.api.loader.ExtensionModelLoadingRequest;
+import org.mule.runtime.module.artifact.activation.internal.ast.MuleSdkExtensionModelLoadingHelper;
+import org.mule.runtime.module.extension.internal.loader.AbstractExtensionModelLoader;
+import org.mule.runtime.module.extension.internal.loader.parser.ExtensionModelParserFactory;
+import org.mule.runtime.module.extension.mule.internal.loader.parser.MuleSdkPluginExtensionModelParserFactory;
+import org.mule.runtime.module.extension.mule.internal.loader.parser.ast.MuleSdkPluginExtensionModelLoadingHelper;
+
+import java.util.Optional;
+import java.util.Set;
+
+import org.slf4j.Logger;
+
+/**
+ * {@link ExtensionModelLoader} implementation for Mule SDK Plugins.
+ *
+ * @since 4.5.0
+ */
+public class MuleSdkPluginExtensionModelLoader extends AbstractExtensionModelLoader {
+
+  private static final Logger LOGGER = getLogger(MuleSdkPluginExtensionModelParserFactory.class);
+
+  @Override
+  public String getId() {
+    return MULE_SDK_EXTENSION_LOADER_ID;
+  }
+
+  @Override
+  protected ExtensionModelParserFactory getExtensionModelParserFactory(ExtensionLoadingContext context) {
+    return new MuleSdkPluginExtensionModelParserFactory();
+  }
+
+  @Override
+  public ExtensionModel loadExtensionModel(ExtensionModelLoadingRequest request) {
```

Comment:
by overriding the entire method you're skipping non-negoatible pieces of logic. Need to extract and override with  finer granularity

Reply:
Overriding only the `doCreate` now, which has a reduced scope.